### ISSUE TITLE
Clarification On Checkout Use

### ISSUE THREAD

**Issue by carsonreinke** [bigcommerce/api#8]
Clarify on how a checkout will use in accordance with an order and cart.  Will a checkout replace an order?  Use an order?  Why will the checkout have an `order_id`?

**Comment by brandonkane** [comment]
This is a key point in allowing us to provide real feedback on the API itself. If the intent here is to build checkouts that then transition into orders, which is how other platforms handle it, then there is a bunch of stuff here that we don't really care about since it relates to payments that have nothing to do with the cart/checkout itself.  If the idea is to have one object that can be used to represent the whole order from cart build to checkout to placed order then there are different things that we'll need as part of the object.

Would be great to add something to the readme to clarify where this new API fits in with the existing BC APIs.  Will hold off on other comments until we get some further clarification here.

**Comment by serroba** [comment]
thanks @carsonreinke and @brandonkane for your feedback. I will try to coordinate this to be clarify in the README.
Our ideation so far, is not replace the order, but provide this concept of checkout that handle the cart plus all the contextual information you need to create an order.

Currently, we create an order when you are in the process of checkout. That order will transition between multiple statuses. So, the idea to expose the `order_id` in the checkout, it's so you can have a relationship between how the cart/checkout was before you have create the order.

In this current approach, we are not considering to merge `checkouts` and `orders` as the same thing. But we would love to get some more examples or use cases, so we can discuss and evaluate our approach.

**Comment by brandonkane** [comment]
Having a separate object to track the pre-order/chckout process is fine, but since the only purpose of that object is to transition into an order, it doesn't make sense to me to have a different structure for it relative to the order itself.  Checkout and Order objects should be nearly identical.  The way they store information about the customer, products in the cart, prices, discounts, etc all needs to be identical or it just creates more work for developers trying to use the API to remember what the differences are.

Order is a superset of the fields on the Checkout, since there are a bunch of things that the Order needs to track that are different from the Checkout.  Things like payment, fulfillment, returns and probably others are only going to be found on an Order, not on the Checkout.  However the structure and content for line items should be identical between the two.  

Having the two in sync makes several other things easier as well. Some of the other comments and issues that have been opened on this repo are about tracking.  If the checkout was not a separate objects but just a different state of an order where it hasn't been placed yet, then any attributes that get on the Checkout will automatically become part of the order, without having to worry about a potentially fragile link between the two via an order_id field and all the extra API calls and overhead on the developers end that are required for that kind of approach.

In short, I think looking at a Checkout API in isolation is short-sighted, if we really want to improve how this process works for developers we need to be thinking holistically about the entire life cycle of an order regardless of the source.

**Comment by carsonreinke** [comment]
@brandonkane well said.  It is confusing to see a checkout and an order and for the most part the properties are far different.  It sounds like this is more then just a "proposed Cart API schema".

**Comment by brandonkane** [comment]
@serroba Haven't seen any updates to the readme and there hasn't been much activity on this from BC since our initial round of questions.  Should we keep monitoring this repository or have you got what you needed?

**Comment by serroba** [comment]
we have plans to update documentation and information. This morning we were reviewing as group the feedback you and others have provided. We will update this soon. I wouldn't recommend to stop the monitoring yet.

**Comment by brandonkane** [comment]
Sounds good, will check back soon for updates.  Looking forward to it!

**Comment by Viralkanz** [comment]
Any update.

**Comment by Viralkanz** [comment]
Any update? @serroba

**Comment by Viralkanz** [comment]
Any update? @serroba

**Comment by serroba** [comment]
@Viralkanz you may find more details on https://developer.bigcommerce.com/api/v3/cart.html
